solidago/src/solidago/mehestan/individual/indiv_matrix_inversion.py

```diff
@@ -0,0 +1,86 @@
+import numpy as np
+import pandas as pd
+
+from .base import IndividualScoresAlgorithm
+
+DEFAULT_ALPHA = 1.0  # Signal-to-noise hyperparameter
+
+
+class MatrixInversionInvididualAlgo(IndividualScoresAlgorithm):
+    def __init__(self, r_max, alpha=DEFAULT_ALPHA):
+        """
+        Initialize in Matrix Inversion algorithm
+
+        Parameters
+        ----------
+        r_max: maximum absolute score of a comparison
+        alpha: Signal-to-noise hyperparameter, regularization term
+        """
+        self.alpha = alpha
+        self.r_max = r_max
+
+    def compute_individual_scores(
+        self, comparison_scores: pd.DataFrame, initial_entity_scores=None
+    ):
+        scores = comparison_scores[["entity_a", "entity_b", "score"]]
+        scores_sym = pd.concat(
+            [
+                scores,
+                pd.DataFrame(
+                    {
+                        "entity_a": scores.entity_b,
+                        "entity_b": scores.entity_a,
+                        "score": -1 * scores.score,
+                    }
+                ),
+            ]
+        )
+
+        # "Comparison tensor": matrix with all comparisons, values in [-R_MAX, R_MAX]
+        r = scores_sym.pivot(index="entity_a", columns="entity_b", values="score")
```

Comment:
out-of-scope:
Would this pivot become a memory bottleneck? 
In my understanding that makes us use O(videos^2) (400M with our current 20k videos), while we could use only O(comparisons) with a sparse matrix. It seems that the below operations would work exactly in the same way.

UPDATE: As it is for individual comparisons, it may be fine. However it opens us to an attack from a user that would make 50k comparisons between 100k videos and crash our server when the ML run allocates >10Gb for them. This is easily feasible with current rate limits